File: bison/Dart/BisonLexerBase.dart
import 'package:antlr4/antlr4.dart';
import 'dart:io';
import 'dart:core';
import 'dart:convert';
import 'dart:collection';
import 'BisonLexer.dart';

abstract class BisonLexerBase extends Lexer
{
    BisonLexerBase(CharStream input) : super(input)
    {
    }

    int percent_percent_count = 0;

    void NextMode()
    {
        ++percent_percent_count;
        if (percent_percent_count == 1)
        {
            return;
        } else if (percent_percent_count == 2)
        {
            this.pushMode(BisonLexer.EpilogueMode);
            return;
        } else
        {
            this.type = BisonLexer.TOKEN_PercentPercent;
            return;
        }
    }

    @override void reset([bool resetInput = false])
    {
	percent_percent_count = 0;
        super.reset(resetInput);
    }
}

File: bison/CSharp/BisonLexerBase.cs
using Antlr4.Runtime;
using Antlr4.Runtime.Misc;
using System.IO;

public abstract class BisonLexerBase : Lexer
{
    private readonly ICharStream _input;
    protected int percent_percent_count;

    public BisonLexerBase(ICharStream input, TextWriter output, TextWriter errorOutput)
            : base(input, output, errorOutput)
    {
        _input = input;
    }


    public void NextMode()
    {
        ++percent_percent_count;
        if (percent_percent_count == 1)
        {
            return;
        } else if (percent_percent_count == 2)
        {
            this.PushMode(BisonLexer.EpilogueMode);
            return;
        } else
        {
            this.Type = BisonLexer.PercentPercent;
            return;
        }
    }

    public override void Reset()
    {
	    percent_percent_count = 0;
	    base.Reset();
    }
}

